lib/matplotlib/legend.py

```diff
@@ -384,8 +384,13 @@ def __init__(self, parent, handles, labels,
         # init with null renderer
         self._init_legend_box(handles, labels, markerfirst)
 
+        # If shadow is activated use framealpha if not
+        # explicitly passed. See Issue 8943
         if framealpha is None:
-            self.get_frame().set_alpha(rcParams["legend.framealpha"])
+            if shadow is True:
+                self.framealpha = 1
+            else:
+                self.get_frame().set_alpha(rcParams["legend.framealpha"])
```

Comment:
why are there two ways of setting framealpha? I would stick with the `self.get_frame().set_alpha()` way, unless we should change it everywhere to `self.framealpha`. Also, we don't need `shadow is True`, it can be just `shadow`.